src/jit/rationalize.cpp

```diff
@@ -116,6 +116,16 @@ void Rationalizer::RewriteSIMDOperand(LIR::Use& use, bool keepBlk)
         addr->gtType = simdType;
         use.ReplaceWith(comp, addr);
     }
+#if defined(_TARGET_X86_)
+    // For x86, we need to handle GT_IND(GT_ADDR(GT_SIMD))
+    else if ((addr->OperGet() == GT_ADDR) && (addr->gtGetOp1()->OperGet() == GT_SIMD))
+    {
+        BlockRange().Remove(tree);
+        BlockRange().Remove(addr);
+
+        use.ReplaceWith(comp, addr->gtGetOp1());
+    }
+#endif // defined(_TARGET_X86_)
```

Comment:
On amd64, we pass vectors by address of temp, so we generate the GT_SIMD node and store the result to a lclVar, then pass the address of the lclVar using putarg_reg. On x86, we pass vectors by value on the stack, so we don't need that temp, and directly have putarg_stk(GT_IND(GT_ADDR(GT_SIMD(...)))).